utils.go

```diff
@@ -0,0 +1,56 @@
+package dpop
+
+import (
+	"crypto"
+	"encoding/hex"
+)
+
+// type MockHash struct {}
+
+// func NewMockHash() hash.Hash {
+// 	return &MockHash{}
+// }
+
+// func (m *MockHash) Write(p []byte) (n int, err error) {
+// 	return 0, nil
+// }
+
+// func (m *MockHash) Sum(b []byte) []byte {
+// 	return nil
+// }
+
+// func (m *MockHash) Reset() {}
+
+// func (m *MockHash) Size() int {
+// 	return 0
+// }
+
+// func (m *MockHash) BlockSize() int {
+// 	return 0
+// }
+
+// func (m *MockHash) Available() bool {
+// 	return false
+// }
+
```

Comment:
Please add comment about what this does

Reply:
If this was supposed to be a different reference let me know, I just removed it as you marked it as dead code in another area.